cgo/lapack.go

```diff
@@ -951,13 +951,15 @@ func (impl Implementation) Dormqr(side blas.Side, trans blas.Transpose, m, n, k
 
 	if lwork == -1 {
 		if left {
-			work[0] = float64(m)
+			work[0] = float64(n)
 			return
 		}
-		work[0] = float64(n)
+		work[0] = float64(m)
 		return
 	}
-
+	if len(work) != lwork {
```

Comment:
(At least at the moment) the rest of the code (and the doc comment) use `if len(work) < lwork`.